### DISCUSSION TITLE
Android Gradle plugin not assigning the correct version classifier

### DISCUSSION THREAD

**Discussion by benedict-lim** [dipien/semantic-version-gradle-plugin#45] [General]
**Describe the bug**
I applied the plugin with Android support.

```
dependencies {
    classpath "com.dipien:semantic-version-android-gradle-plugin:1.4.1"
}

version = "1.0.0"
apply plugin: "com.dipien.android.semantic-version"
```

I do not want the `-SNAPSHOT` version classifier in my build version name, so I declared the following extension.

```
semanticVersion {
    snapshot = false
}
```

However, the `-SNAPSHOT` version classifier is still being applied. I tried other properties such as `alpha`, `beta` and `versionClassifier` itself, but I could not successfully apply any of them.

**To Reproduce**
1. Apply plugin for Android support
2. Set `snapshot` property to `false`
3. Build Android project
4. Check `VERSION_NAME` in the generated `BuildConfig.java` file

**Expected behavior**
Version classifier should be applied according to the properties set in the extension.

**Screenshots**

**Additional context**
Although I have not tested the code locally, I thought that the following code lines might be the ones causing the issue.

https://github.com/dipien/semantic-version-gradle-plugin/blob/92f6b30f9c134b434401d44ebdb7a45bed019cea/semantic-version-android-gradle-plugin/src/main/java/com/semanticversion/android/gradle/plugin/SemanticVersionAndroidGradlePlugin.kt#L47-L48

Although `versionCode` is created with config, the creation of `versionName` does not take config into account.

By the way, I tried setting the Android-specific property in the extension, and it worked. The base properties such as `snapshot`, `alpha`, `beta` etc. do not.

```
semanticVersion {
    minSdkVersionAsVersionCodePrefix = true
}
```

**Comment by maxirosson** [comment]
The problem is that the plugin is trying to read the extension before the plugin is applied, so the extension is still with null values. This is a bad design of the plugin, that I am going to correct on the next version.

Meanwhile, you can do the following workaround:

```
ext.snapshot = false
apply plugin: "com.dipien.semantic-version"
```

**Comment by benedict-lim** [comment]
I see... I appreciate the explanation. I thought that I might be using the plugin in a wrong way, so it is good to know the root cause of the problem.

Thank you for providing the workaround! I will try to work with that until the next version is released.
